src/sentry/migrations/0380_backfill_monitor_env_initial.py

```diff
@@ -0,0 +1,82 @@
+# Generated by Django 2.2.28 on 2023-03-09 23:03
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+DEFAULT_ENVIRONMENT_NAME = "production"
+
+
+def backfill_monitor_environments(apps, schema_editor):
+    Project = apps.get_model("sentry", "Project")
+    Monitor = apps.get_model("sentry", "Monitor")
+    Environment = apps.get_model("sentry", "Environment")
+    EnvironmentProject = apps.get_model("sentry", "EnvironmentProject")
+    MonitorEnvironment = apps.get_model("sentry", "MonitorEnvironment")
+
+    queryset = RangeQuerySetWrapperWithProgressBar(
+        Monitor.objects.filter(monitorenvironment__isnull=True).values_list(
+            "id", "organization_id", "project_id", "status", "next_checkin", "last_checkin"
+        ),
```

Comment:
`Monitor` has a `status` field. Let's filter to only monitors with `status not in (1, 2, 3)` https://github.com/getsentry/sentry/blob/2ec9adaa218e4097caa36e456c160b7cbe906b94/src/sentry/constants.py#L453-L455.

Not 100% sure about the hidden/disabled status

Reply:
those aren't used, the `DELETION` ones are. it was just subclassed and grabbed the extra statuses

Reply:
`DISABLED` we do use

Reply:
I can see all the statuses in use in the database. Probably old rows, but it might be safer to not try update them in case we run into other weird edge cases.

Reply:
I would consider these to be all non-terminal states
https://github.com/getsentry/sentry/blob/b1fa1ebd57e42a7a42d770b6050c48d0efb554d9/src/sentry/monitors/endpoints/organization_monitors.py#L31

Reply:
Ok, so maybe `status not in (2, 3)` then, since disabled is valid

Reply:
Then we can clean up monitors with status (2, 3) and where project doesn't exist in a later migration